mailet/api/src/main/java/org/apache/mailet/LoopPrevention.java

```diff
@@ -84,6 +84,14 @@ public RecordedRecipients merge(MailAddress... other) {
             return merge(ImmutableSet.copyOf(other));
         }
 
+        public RecordedRecipients mergeIfEmpty(MailAddress... other) {
```

Comment:
Why merge only if empty? Why not merge all the time?

Reply:
for example, we setup to make A forward mail to B, B to C, C to A.

someone send email to A, then james receive an email with A as recipient, we have to add A to the recorded recipient list (before adding, the list is empty)
when james receive an email with B as recipient, adding B to the recorded recipient list is not necessary (before adding, the list has already included A and B)

Reply:
Isn't it managed naturrally by the `Set` data structure?

Reply:
if I just merge it without checking empty, there would be two issues:
- when others read code, it could make them confused because the originalRecipient is always merged. By using mergeIfEmpty, they could understand how this actually work.
- create unnecessary operations. It does not impact performance much but it is good if we can avoid.